### DISCUSSION TITLE
Dotty highlighting for Scala sources?

### DISCUSSION THREAD

**Discussion by armanbilge** [planet42/Laika#268] [Q&A] [answered]
Currently, Scala 3 "Dotty" highlighting is activated only if the sources are marked as `dotty`:
```
```dotty
```

In my case, the problem is that this doesn't work with mdoc:
```
```dotty mdoc
```

I'm not sure if there's any good way to resolve this, without sacrificing either Scala 2 or Scala 3 syntax highlighting.

**Comment by jenshalm** [comment, marked as answer]
Yes, there is no obvious default, as everyone would have different preferences. I kept the Scala 3 syntax separate for two reasons:

1. The nearly non-existent adoption of Scala 3

2. If they are initially separate, users can still re-configure them to get Scala 3 syntax for a `scala` code block, the other way round would not be possible. If they are pre-merged you could not "configure them apart".

If you want to switch today, you can do this:

```scala
laikaExtensions := Seq(SyntaxHighlighting.withSyntaxBinding("scala", DottySyntax))
```

**Reply by armanbilge** [reply]
Aha, I had a feeling there must be a config for this. Thanks as always!

**Reply by armanbilge** [reply]
Although, I think that as more projects cross-build for Scala 3 they will be avoiding the Scala 3 keywords. So not sure when, but at some point it's probably okay to flip the switch.

**Reply by jenshalm** [reply]
At some point, yes. How about "when Spark is on Scala 3" 🙂

**Reply by armanbilge** [reply]
If anyone else stumbles here, the method is actually called `withSyntaxBinding` :)

Btw, sbt-typelevel-site will apply the extension automatically if you are using Scala 3.
- https://github.com/typelevel/sbt-typelevel/pull/219
